### DISCUSSION TITLE
Titles and separate files

### DISCUSSION THREAD

**Discussion by solernou** [Witiko/markdown#381] [Q&A] [answered]
I have different files for different sections and subsections, and this works well using in a ` hybrid ` `markdown` environment. However, hybrid environments don't seem capable of rendering the headers correctly. 

Here's a minimal example to illustrate the issue. Type this into `document.tex`:

    \documentclass{article}
    \usepackage[headerAttributes,contentBlocks]{markdown}
    \begin{document}
    \begin{markdown}
    Plain
    ====
    $\sqrt{-1}$ *equals* $i$

    /structure.md

    \end{markdown}

    \begin{markdown*}{hybrid}
    Hybrid
    ======
    $\sqrt{-1}$ *equals* $i$
    \input{structure.md}
    \end{markdown*}

    \end{document}

Now, type the following into `structure.md`:

    Intro {intro}
    ------

    There is a lovely Markdown CTAN package.

    Section 1
    ----------

    \input{section1.md}

And the following into `section1.md`:

    I work on lots of files at the same time.

Running ` lualatex --shell-escape document.tex` on an up-to-date Ubuntu 22.04 using the CTAN package: markdown 2022/01/30 v2.13.0-0-g1f680a8 markdown renderer I get the attached pdf: 

  [document.pdf](https://github.com/Witiko/markdown/files/13649887/document.pdf)

which on the "plain" `markdown` environment:
  - renders the headers correctly
  -  fails to include all the files
  
 and on the "hybrid" `markdown` environment:
  - fails to render the headers
  - correctly inputs all the files.
  

Any help?

**Comment by Witiko** [comment, marked as answer]
Hello and thanks for using the Markdown package.

The `\input` command includes .tex files, not .md files. Therefore, using `\input{structure.md}` instructs TeX to interpret the file `structure.md` as if it were a .tex file, not a .md file, which will fail to render the headings (or any other markdown elements) correctly.

If you'd like to include .md files in you documents, you can either:

1. Use file transclusion to include the file `section1.md` from the file `structure.md` similarly to how you included the file `section.md` from `document.tex`:

        /section1.md

2. Use the command `\markdownInput` instead of `\input` either outside the `markdown` environment or inside the markdown environment with the `hybrid` option enabled:

        \markdownInput{structure.md}

    

        \markdownInput{section1.md}

I hope this helps.

**Reply by solernou** [reply]
Thanks @Witiko, that did the trick! And thanks for the package, it is making my day. I wonder if this could be clarified in the docs.

**Reply by Witiko** [reply]
Happy to help! In commit 622ee5620cd812386c51f8ddffded891524d7697 from https://github.com/Witiko/markdown/pull/382, I updated the technical documentation and the user manual, so that they mention both `\markdownInput` and `\input` to improve discoverability. In the user manual, the `\markdownInput` macro is now used in the first examples to make it harder to miss. Furthermore, this discussion thread will also make it easier for users to find out about the `\markdownInput` macro, thanks for creating it.

**Reply by solernou** [reply]
Thanks, that's ace. Feel free to close the discussion.
